inc/ThirdParty/Hostings/Wpengine.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace WP_Rocket\ThirdParty\Hostings;
+
+use WP_Rocket\ThirdParty\ReturnTypesTrait;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+use WpeCommon;
+
+/**
+ * Compatibility class for WP Engine.
+ *
+ * @since 3.6.1
+ */
+class Wpengine implements Subscriber_Interface {
+	use ReturnTypesTrait;
+
+	/**
+	 * Array of events this subscriber wants to listen to.
+	 *
+	 * @since 3.6.1
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		if (
+		! (
+			class_exists( 'WpeCommon' )
+			&&
+			function_exists( 'wpe_param' )
+		)
+		) {
+			return [];
+		}
+
+		return [
+			'rocket_varnish_field_settings'           => 'varnish_field',
+			'rocket_display_input_varnish_auto_purge' => 'return_false',
+			'rocket_cache_mandatory_cookies'          => [ 'return_empty_array', PHP_INT_MAX ],
+			'rocket_advanced_cache_file'              => 'return_empty_string',
+			'admin_init'                              => [
+				[ 'remove_notices' ],
+				[ 'run_rocket_bot_after_wpengine' ],
+			],
+			'set_rocket_wp_cache_define'              => 'return_true',
+			'do_rocket_generate_caching_files'        => 'return_false',
+			'after_rocket_clean_domain'               => 'clean_wpengine',
+			'rocket_buffer'                           => [ 'add_footprint', 50 ],
+			'rocket_disable_htaccess'                 => 'disable_htaccess',
+		];
+	}
+
+	/**
+	 * Changes the text on the Varnish one-click block.
+	 *
+	 * @since 3.6.1
+	 *
+	 * @param array $settings Field settings data.
+	 *
+	 * @return array modified field settings data.
+	 */
+	public function varnish_field( $settings ) {
```

Comment:
How about using a more descriptive method name here? `varnish_field` doesn't tell what this method is doing.

Maybe `varnish_addon_title` or something like that instead?